src/blob.rs

```diff
@@ -426,12 +432,63 @@ impl<'a> BlobObject<'a> {
         })?;
         let orientation = self.get_exif_orientation(context);
 
-        let do_scale = img.width() > img_wh || img.height() > img_wh;
+        fn exceeds_bytes(
+            context: &Context,
+            img: &DynamicImage,
+            max_bytes: Option<usize>,
```

Comment:
I'd rather change this to simply `u32`, the function does nothing when it's `None`. You can use `max_bytes.map_or_else(false, exceeds_bytes)` to call it.

Reply:
It has to be sth. like `max_bytes.map_or_else(false, |max| exceeds_bytes(context, &img, max, encoded)`. I can still do this, of course, if you say it's better.

BTW the fn is called `encode_img_exceeds_bytes()` now as it doesn't only count bytes anymore. If we want to avoid this, we could replace
```
encode_img_exceeds_bytes(context, &img, max_bytes, &mut encoded)
``` 
by sth. like 
```
max_bytes.map_or_else(false, |max| {encode_img(&img, &mut encoded); exceeds_bytes(context, max, encoded)}
```
splitting encode_img_exceeds_bytes() into two function calls.